### DISCUSSION TITLE
The avalonia android program cannot be called the 'InvokeCoreAsync' method if the connection is not active.

### DISCUSSION THREAD

**Discussion by IftikharK** [AvaloniaUI/Avalonia#14609] [Q&A]
I have cross-platform application and have tried to communicate with the SignalR server in Avalonia desktop, browser and android program.
It works fine with desktop and browser program but with android,  it throw an exception "The 'InvokeCoreAsync' method cannot be called if the connection is not active".  Don't understand why same lines of code working with desktop and browser but not in android. Any help would be greatly appreciated. 
```csharp
using Avalonia.Threading;
using Avalonia.XPlat.Constants;
using Avalonia.XPlat.Interfaces;
using Avalonia.XPlat.Models;
using Avalonia.XPlat.Services;
using Microsoft.AspNetCore.SignalR.Client;
using ReactiveUI;
using System;
using System.Collections.ObjectModel;
using System.Reactive;
using System.Threading.Tasks;

namespace Avalonia.XPlat.ViewModels
{	
    public class ChatViewModel : ViewModelBase
    {
		private readonly HubConnection? _hubConnection;
        
		private string? _userName;
        private string? _message;
        private string? _viewTitle;
        private ObservableCollection<string?> _messagesList;

        public string? ViewTitle
        {
            get => _viewTitle;
            set => this.RaiseAndSetIfChanged(ref _viewTitle, value);
        }

        public string? UserName
        {
            get => _userName;
            set => this.RaiseAndSetIfChanged(ref _userName, value);
        }

        public string? Message
        {
            get => _message;
            set => this.RaiseAndSetIfChanged(ref _message, value);
        }

        public ObservableCollection<string?> MessagesList
        {
            get => _messagesList;
            set => this.RaiseAndSetIfChanged(ref _messagesList, value);
        }

        public ReactiveCommand<Unit, Unit> ConnectCommand { get; }
        public ReactiveCommand<Unit, Unit> SendCommand { get; }

		private const string BaseUrl = "https://localhost:7256";
		private const string hub = "/chatHub";
		
        public ChatViewModel()
        {
            ViewTitle = "Real-Time Chat!";

            _messagesList = new ObservableCollection<string?>();

			 _hubConnection = new HubConnectionBuilder()
				 .WithUrl($"{BaseUrl}/{hub}")
				 .Build();
				 
            _hubConnection.On<ChatMessage>(
                "ReceiveMessage",
                (message) =>
                {
                    Dispatcher.UIThread.InvokeAsync(() =>
                    {
                        var newMessage = $"{message.User}: {message.Message}";
                        MessagesList.Add(newMessage);
                    });
                }
            );

            SendCommand = ReactiveCommand.Create(SendMessage);
        }

        private async void SendMessage()
        {
            try
            {
				await _hubConnection!.StartAsync();
                
				var chatMessage = new ChatMessage(UserName, Message);
                
				await _hubConnection!.InvokeAsync("SendMessage", chatMessage);
            }
            catch (Exception ex)
            {
                MessagesList.Add(ex.Message);
            }
        }       
    }
}
```

**Comment by maxkatz6** [comment]
I am not sure if it's related to Avalonia. Could be something related to websockets implementation on android. Hard to tell from this description.

**Comment by IftikharK** [comment]
thanks for response. If it's related to websockets implementation then it should also not work with desktop and browser.

**Comment by Gillibald** [comment]
This is not related to Avalonia. Please have a look at the debug output.

**Reply by IftikharK** [reply]
I have checked the debug output and found the following error details. In hubconnection I am using this url: https://localhost:7256  Anyone have idea how to fix the following error?

{System.Net.Http.HttpRequestException: Connection failure
 ---> Java.Net.ConnectException: Failed to connect to localhost/127.0.0.1:7256
   at Java.Interop.JniEnvironment.InstanceMethods.CallVoidMethod(JniObjectReference instance, JniMethodInfo method, JniArgumentValue* args) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/obj/Release/net7.0/JniEnvironment.g.cs:line 20370
   at Java.Interop.JniPeerMembers.JniInstanceMethods.InvokeAbstractVoidMethod(String encodedMember, IJavaPeerable self, JniArgumentValue* parameters) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/Java.Interop/JniPeerMembers.JniInstanceMethods_Invoke.cs:line 47
   at Javax.Net.Ssl.HttpsURLConnectionInvoker.Connect() in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net8.0/android-34/mcw/Javax.Net.Ssl.HttpsURLConnection.cs:line 443
   at Xamarin.Android.Net.AndroidMessageHandler.<>c__DisplayClass133_0.<ConnectAsync>b__0() in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Xamarin.Android.Net/AndroidMessageHandler.cs:line 526
  --- End of managed Java.Net.ConnectException stack trace ---

**Comment by timunie** [comment]
making this a discussion for now as we think it's not related to Avalonia. If it points out it is, we can still convert it into a bug report again.

**Reply by kutarai** [reply]
In Android emulator uses 10.0.2.2 for localhost. So if your SignalR server is on localhost, the android app should use 10.0.2.2 to connect to it.
